packages/common/src/api/GqlContext.tsx

```diff
@@ -1,14 +1,83 @@
 import React, { FC, useMemo, useEffect, useState, useCallback } from 'react';
 import { createContext } from 'react';
-import { GraphQLClient } from 'graphql-request';
+import {
+  BatchRequestDocument,
+  GraphQLClient,
+  RequestDocument,
+  Variables,
+} from 'graphql-request';
+import { AuthError } from '../authentication/AuthContext';
+import { LocalStorage } from '../localStorage';
 
-export const createGql = (url: string): { client: GraphQLClient } => {
-  const client = new GraphQLClient(url, { credentials: 'include' });
+class GQLClient extends GraphQLClient {
+  private client: GraphQLClient;
+  private emptyData: object;
+
+  constructor(url: string, options?: RequestInit) {
+    super(url, options);
+    this.client = new GraphQLClient(url, options);
+    this.emptyData = {};
+  }
+
+  public rawRequest<T, V = Variables>(
+    query: string,
+    variables?: V,
+    requestHeaders?: RequestInit['headers']
+  ): Promise<{
+    data: T;
+    extensions?: any;
+    headers: Headers;
+    status: number;
+  }> {
+    return this.client.rawRequest(query, variables, requestHeaders);
+  }
+
+  public request<T, V = Variables>(
+    document: RequestDocument,
+    variables?: V,
+    requestHeaders?: RequestInit['headers']
+  ): Promise<T> {
+    const response = this.client.request(document, variables, requestHeaders);
+    // returning an empty object in order to give the caller a stable reference
+    // without it, the page will re-render continuously
+    return response.then(
+      data => data ?? this.emptyData,
+      ({ response }) => {
+        if (response && response.errors) {
+          if (
+            response.errors.some(
+              ({ message }: { message?: string }) =>
+                message === 'Unauthenticated'
+            )
+          ) {
+            LocalStorage.setItem('/auth/error', AuthError.Unauthenticated);
+          }
+        }
+        return this.emptyData;
```

Comment:
as noted.. the reason that internal orders were endlessly rendering was because the gql response was returning `null` for `data` which can't be used as a stable reference for the component.

Instead, create an object in the class and return that, so it is stable.